src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionService.swift

```diff
@@ -321,8 +326,11 @@ class ENAExposureSubmissionService: ExposureSubmissionService {
 					return
 				}
 				switch testResult {
-				case .negative, .positive, .invalid:
+				case .positive, .negative, .invalid:
 					self.store.testResultReceivedTimeStamp = Int64(Date().timeIntervalSince1970)
+					self.store.testResultDate = Date()
```

Comment:
in the line above, we already store the data (in another format). You could use this timestamp after you converted it back to `Date` for your use.
At the end, we would not save the same data twice.

Reply:
Good point.